pkg/remoteresolution/cache/cache.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2024 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"sort"
+	"strconv"
+	"time"
+
+	"context"
+
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	utilcache "k8s.io/apimachinery/pkg/util/cache"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	// DefaultMaxSize is the default size for the cache
+	DefaultMaxSize = 1000
+
+	// ConfigMapName is the name of the ConfigMap containing cache configuration
+	ConfigMapName = "resolver-cache-config"
+
+	// ConfigMapNamespace is the namespace of the ConfigMap
+	ConfigMapNamespace = "tekton-pipelines-resolvers"
+)
+
+var (
+	// DefaultExpiration is the default expiration time for cache entries
+	DefaultExpiration = 5 * time.Minute
+)
+
+// ResolverCache is a wrapper around utilcache.LRUExpireCache that provides
+// type-safe methods for caching resolver results.
+type ResolverCache struct {
+	cache  *utilcache.LRUExpireCache
+	logger *zap.SugaredLogger
+}
+
+// NewResolverCache creates a new ResolverCache with the given expiration time and max size
+func NewResolverCache(maxSize int) *ResolverCache {
+	return &ResolverCache{
+		cache: utilcache.NewLRUExpireCache(maxSize),
+	}
+}
+
+// InitializeFromConfigMap initializes the cache with configuration from a ConfigMap
+func (c *ResolverCache) InitializeFromConfigMap(configMap *corev1.ConfigMap) {
+	// Set defaults
+	maxSize := DefaultMaxSize
+	ttl := DefaultExpiration
+
+	if configMap != nil {
+		// Parse max size
+		if maxSizeStr, ok := configMap.Data["max-size"]; ok {
+			if parsed, err := strconv.Atoi(maxSizeStr); err == nil && parsed > 0 {
+				maxSize = parsed
+			}
+		}
+
+		// Parse default TTL
+		if ttlStr, ok := configMap.Data["default-ttl"]; ok {
+			if parsed, err := time.ParseDuration(ttlStr); err == nil && parsed > 0 {
+				ttl = parsed
+			}
+		}
+	}
+
+	c.cache = utilcache.NewLRUExpireCache(maxSize)
+	DefaultExpiration = ttl
+}
+
+// InitializeLogger initializes the logger for the cache using the provided context
+func (c *ResolverCache) InitializeLogger(ctx context.Context) {
+	if c.logger == nil {
+		c.logger = logging.FromContext(ctx)
+	}
+}
```

Comment:
Bug: 
This will leak state. The ResolverCache is a global object and should not include the context from the resolvers `Resolve()` call. I suggest providing the context to every call like `cache.GetGlobalCache().Get(ctx, key)` or by allowing the context to decorate the global cache along these lines:
```suggestion
func  WithLogger(logger *zap.SugaredLogger) c *ResolverCache {
	return ResolverCache{logger: logger, cache: GetGlobalCache().cache}
}
```